robustcontrol/utils.py

```diff
@@ -1910,6 +1928,45 @@ def state_controllability(A, B):
 
     return state_control, u_p, control_matrix
 
+def state_observability(A, C):
+    """
+    This method checks if the state space description of the system is state
+    observ able according to Definition 4.2.
+
+    Parameters
+    ----------
+    A : numpy matrix
+        Matrix A of state-space representation.
+    C : numpy matrix
+        Matrix C of state-space representation.
+
+    Returns
+    -------
+    state_obv : boolean
+        True if state observable
+    y_p : array
+        Ouput pole vectors for the states y_p_i
+    obs_matrix : numpy matrix
+        State Observability Matrix
+    v_r : numpy array
+        Right Eigenvectors
+
+      """
+    state_obs = True
+
+    # Compute output pole vectors.
+    ev, vr = scipy.linalg.eig(A, left=False, right=True)
+    y_p = []
+    for i in range(vr.shape[1]):
+        vri = np.asmatrix(vr[:, i])
+        y_p.append(C*vri.T)
+    state_obs = not any(np.linalg.norm(x) == 0.0 for x in y_p)
```

Comment:
Checking for equality is usually a bad idea in floating point - should rather be something like `< small_number` instead of `== 0`.

Reply:
Fixed